tools/parse/src/main/scala/org/allenai/nlpstack/parse/poly/eval/ParseEvaluator.scala

```diff
@@ -124,6 +124,72 @@ case object UnlabeledBreadcrumbAccuracy extends ParseStatistic {
   }
 }
 
+/** CposAccuracy stores the statistics necessary to compute coarse-part-of-speech tagging
+  * accuracy, which is the percentage of tokens assigned the correct CPOS by the parser.
+  *
+  * @param verbose if true, reporting gives more detailed error analysis of the tagging mistakes
+  */
+case class CposAccuracy(verbose: Boolean = false) extends ParseStatistic {
+  var numCorrect = 0
+  var numTotal = 0
+  var numParses = 0
+  var errorHistogram = Map[(Symbol, Symbol), Int]()
+
+  override def notify(candidateParse: Option[PolytreeParse], goldParse: PolytreeParse): Unit = {
+    numParses += 1
+    numTotal += goldParse.breadcrumb.tail.size
+    candidateParse match {
+      case Some(candParse) =>
+        if (candParse.breadcrumb.size == goldParse.breadcrumb.size) {
+          numCorrect +=
+            candParse.tokens.tail.zip(goldParse.tokens.tail) count {
+              case (candToken, goldToken) =>
+                candToken.getDeterministicProperty('cpos) == goldToken.getDeterministicProperty('cpos)
+            }
+          candParse.tokens.tail.zip(goldParse.tokens.tail) filter {
+            case (candToken, goldToken) =>
+              candToken.getDeterministicProperty('cpos) != goldToken.getDeterministicProperty('cpos)
+          } foreach {
+            case (candToken, goldToken) =>
+              val candCpos = candToken.getDeterministicProperty('cpos)
+              val goldCpos = goldToken.getDeterministicProperty('cpos)
+              errorHistogram = errorHistogram.updated(
+                (candCpos, goldCpos),
+                1 + errorHistogram.getOrElse((candCpos, goldCpos), 0)
+              )
+          }
+        } else { // skip the parse if the tokenization is different
+          println(s"WARNING -- Skipping parse: ${candParse.sentence.asWhitespaceSeparatedString}" +
+            s" tokenized differently than gold: ${goldParse.sentence.asWhitespaceSeparatedString}")
+          numParses -= 1
```

Comment:
Mainly because I still want to increment those counts if there's a failed parse (the next else clause), so it's a choice between incrementing in two of the three blocks, or incrementing outside and decrementing in one of the blocks. Perhaps it's less confusing to do the two increments instead?